### PR TITLE
[SPARK-31532][SPARK-31234][SQL][2.4][FOLLOWUP] Use lowercases for GLOBAL_TEMP_DATABASE config in SparkSessionBuilderSuite

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This PR intends to fix test code for using lowercases for the `GLOBAL_TEMP_DATABASE` config in `SparkSessionBuilderSuite`. The handling of the config is different between branch-3.0+ and branch-2.4. In branch-3.0+, Spark always lowercases a value in the config, so I think we had better always use lowercases for it in the test.

This comes from the @dongjoon-hyun 
 comment: https://github.com/apache/spark/pull/28316#issuecomment-619303160

### Why are the changes needed?

To fix the test failure in branch-2.4.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Fixed the test.

### CODE REVIEW

**Comment by maropu** [comment]
cc: @dongjoon-hyun @cloud-fan @yaooqinn

**Comment by dongjoon-hyun** [comment]
Thank you, @maropu !

**Review by yaooqinn — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala [hunk 1]
```diff
@@ -157,28 +157,28 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach {
     "spark context conf values in SessionState") {
     val session = SparkSession.builder()
       .master("local")
-      .config(GLOBAL_TEMP_DATABASE.key, value = "globalTempDB-SPARK-31234")
+      .config(GLOBAL_TEMP_DATABASE.key, value = "globaltempdb-spark-31234")
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:160]: Now, I understand why you change the conf. +1.

Comment (maropu) [reply sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala:160]: Need this one? Dropping this looks fine to me though.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you for the swift fix.
I ran the test locally. I'll merge this to recover `branch-2.4`.

**Comment by maropu** [comment]
Thanks for the quick response, @dongjoon-hyun !

**Pull request closed without merging**